ai/core/src/unitycatalog/ai/core/envs/base.py

```diff
@@ -0,0 +1,41 @@
+import json
+import os
+
+
+class _EnvironmentVariable:
+    def __init__(self, name: str, type_: type, default_value: str, description: str):
+        self.name = name
+        self.type = type_
+        self.default_value = str(default_value) if type_ is str else default_value
+        self.description = description
+
+    def _get_raw(self) -> str:
+        return os.getenv(self.name)
+
+    def get(self) -> any:
+        raw_val = self._get_raw()
+        if raw_val is not None:
+            try:
+                if self.type is list:
+                    try:
+                        return json.loads(raw_val)
+                    except Exception:
+                        return [x.strip() for x in raw_val.split(",")]
```

Comment:
This seems a bit weird, if we have to support list[str] can we only support list[str] and not list of any other types?

Reply:
Yeah, let's just support casting collection types properly.